### ISSUE TITLE
feat(ml): score out-of-distribution inputs by embedding distance so rejectedOod has a producer

### ISSUE THREAD

**Issue by LukeSantossz** [LukeSantossz/visiosoil-app#194]
## Description

`ClassificationOutcome.rejectedOod` is a reserved status with no producer and two candidate shapes, one of which needs a dataset nobody has costed and the other of which cannot work. There is a third: score the distance from the classifier's own penultimate embedding to the training classes. It needs no negative-class training set, no second model artifact, and no collection campaign.

## Context

Found by the architecture review of 2026-08-25.

`soil-classification.md` §24 question 5 asks whether the negative class lives in the model or is handled by the quality gate plus a threshold, and ADR 0015 declares the enum member while recording that it has no producer and that E12 informs the choice.

**The second candidate cannot work, and this is worth stating plainly because the question is currently posed as a two-way choice.** A photograph of a wall is well exposed, in focus, high in contrast, free of specular hotspots and above the effective-resolution floor. It passes all seven acceptance criteria in `ml/src/image_quality.py` and is then assigned one of five soil classes with whatever confidence the softmax produces. The gate is photometric and geometric; being not-soil is semantic, and no threshold on blur or luminance reaches it. `soil-classification.md` §5 already records the negatives gap as Total.

**The first candidate is expensive in the one currency the project is short of.** A sixth trained class needs a non-soil photograph set large enough to train on, and ADR 0014's whole argument for the archive is that the dataset costs rig time and zero laboratory analyses. A negative-class campaign reintroduces exactly the collection cost that decision removed.

**One-class scoring is the standard answer when the anomalous examples are rare, expensive to label, or impossible to enumerate**, which is the case here: the complement of soil is not a class, it is everything else. Training uses only normal data and inference scores deviation against a threshold. The image-space form — an autoencoder scoring reconstruction error — is a poor fit for the same reason VAEs were rejected in ADR 0010: it needs a second model and reconstructing high-frequency texture is what that family does worst. The **feature-space** form costs almost nothing, because the embedding is already computed on the way to the softmax.

## Current State

**The embedding exists and is discarded.** `ml/src/model.py:59-63`:

```python
x = keras.layers.GlobalAveragePooling2D(name="gap")(x)
x = keras.layers.BatchNormalization(name="bn")(x)
x = keras.layers.Dense(256, activation="relu", name="dense_head")(x)
x = keras.layers.Dropout(dropout, name="dropout")(x)
outputs = keras.layers.Dense(num_classes, activation="softmax", name="predictions")(x)
```

The model exposes one output. `ml/src/export.py:159-172` declares a single output of shape `[1, num_classes]`.

**The app cannot express the outcome yet.** `InferenceService.classify` returns `InferenceResult?`. SPEC 0035 and ADR 0015 replace that with a report carrying `ClassificationOutcome`, and `rejectedOod` is declared there with no producer. This issue is what would give it one; it does not change the enum.

**Nothing is calibrated and nothing can be.** `ml/src/evaluate.py` discards the probability vector at `:60` and never touches intermediate activations.

## Proposed Solution

Expose the embedding, fit a class-conditional score on the training split, publish the score's parameters in `spec.json`, and threshold it in the app.

1. **Export a second output tensor** carrying the `dense_head` activation. `spec.json`'s `output` section gains a second entry, coordinated with #187 which is already opening that schema.
2. **Fit on the training split only**, after the final model is frozen and quantized — same ordering argument as #187, since quantization moves the activations as well as the logits.
3. **Choose a score that fits in the contract.** This is the constraint that decides the shape, and it is arithmetic rather than preference:

   | Score | Parameters to ship | Verdict |
   |---|---|---|
   | Full-covariance Mahalanobis, 256-d | 5 means plus a 256×256 inverse covariance, ~65k floats | **No.** Roughly 700 KB of JSON, and a full covariance is not estimable from a few hundred training groups |
   | Shared diagonal covariance, 256-d | 5 means plus 256 variances, ~1.5k floats | Viable, ~20 KB |
   | PCA to 32-d, then full covariance | Projection, 5 means, 32×32 inverse, ~9k floats | Viable, and better conditioned |
   | kNN distance | A stored subsample of training embeddings | Rejected: the artifact grows with the dataset, and it is the option that most resembles shipping the training set |

   Shared diagonal is the default; PCA is the fallback if the diagonal assumption measures badly.
4. **Add experiment E16**: score a small non-soil set and the images the quality gate rejected, report the separation, and sweep the threshold at declared false-positive rates. Refusing a genuine soil photograph is the expensive error here, so the operating point is chosen against that.
5. **Degrade honestly.** If the parameters are absent from `spec.json`, no score is computed and `rejectedOod` keeps having no producer, which is the status quo rather than a failure. A missing optional field must not become a silent fallback, per ADR 0015.

Criteria for this shape: it adds no second artifact to version, convert and verify, which is the cost ADR 0008 and ADR 0009 both refused to pay twice; the fit needs only soil, which is the dataset that exists; the evaluation set is a morning's photography rather than a campaign; and the parameters live in the file that already carries every other per-class constant.

**Two limits, and the second one matters more than it looks.**

The wiki's coverage of the classical one-class family is by application rather than from a primary text and is marked as needing expansion, so treat the mechanism as well supported and the specific estimator choice as unvalidated.

**Deviation scoring is structurally blind to what is abnormally wrong but distributionally normal.** The case that matters here is moist unsieved field material: it is soil, it will sit near the training manifold on most of what a soil-trained embedding measures, and it is precisely the input ADR 0014 says the model has no evidence about. So this is a useful signal for gross non-soil inputs and **it is not a solution to R11**. Anything that reads it as one would be claiming the bench-to-field gap is closed by a threshold.

## Dependencies

- **SPEC 0035 / #79** — the outcome enum and the report type have to exist before anything can produce `rejectedOod`.
- **#187** — the schema change and the fit-after-quantization ordering.
- **#188** — the evaluation harness that would compute and sweep the score.
- Informs `soil-classification.md` §24 question 5, which currently poses a two-way choice.
- Related to E12, which ADR 0015 names as the experiment informing this decision.

## Acceptance Criteria

- [ ] The exported artifact carries the penultimate activation as a second output, and `spec.json` declares it.
- [ ] The score's parameters are fitted on the training split only, against the frozen quantized artifact, and are published in `spec.json` within a stated size budget.
- [ ] E16 is recorded in §11, reporting separation against a non-soil set and against quality-gate rejects, with the threshold swept at declared false-positive rates.
- [ ] A missing or malformed score section produces no score and no `rejectedOod`, never a fallback threshold. A test asserts it.
- [ ] `soil-classification.md` §24 question 5 lists three options rather than two, and records that the quality-gate-plus-threshold option cannot detect a semantic mismatch.
- [ ] The record states, in the same place the score is documented, that this does not detect field-fresh material.
